### PR TITLE
Add references to the bibliography and a cross reference in open source software

### PR DESCRIPTION
**Pull request by RichardJActon**
### Summary

Add 2 references mentioned in the text but not present in the bibliography to the bibliography, and add a cross reference to open source licensing chapter where it was mentioned here.

Addresses some points raised in #3705 

### List of changes proposed in this PR (pull-request)

* cross reference licensing chapter on first mention of 'open-source licence'
* Add reference to Gorgolewski et al 2016 to the bibliography
* Add reference to Pedregosa et al. 2012 to the bibliography

### What should a reviewer concentrate their feedback on?

- [ ] Everything looks ok?

### Acknowledging contributors

- [x] All contributors to this pull request are already named in the [table of contributors](https://github.com/the-turing-way/the-turing-way/blob/main/README.md#contributors) in the README file.

### CODE REVIEW

File: book/website/_bibliography/references.bib [hunk 1]
```diff
@@ -2662,3 +2662,41 @@ @misc{WEF2022commons
...
+  doi = {10.1038/sdata.2016.44},
+  url = {https://www.nature.com/articles/sdata201644},
+  urldate = {2024-06-21},
+  abstract = {Abstract             The development of magnetic resonance imaging (MRI) techniques has defined modern neuroimaging. Since its inception, tens of thousands of studies using techniques such as functional MRI and diffusion weighted imaging have allowed for the non-invasive study of the brain. Despite the fact that MRI is routinely used to obtain data for neuroscience research, there has been no widely adopted standard for organizing and describing the data collected in an imaging experiment. This renders sharing and reusing data (within or between labs) difficult if not impossible and unnecessarily complicates the application of automatic pipelines and quality assurance protocols. To solve this problem, we have developed the Brain Imaging Data Structure (BIDS), a standard for organizing and describing MRI datasets. The BIDS standard uses file formats compatible with existing software, unifies the majority of practices already common in the field, and captures the metadata necessary for most common data processing operations.},
+  langid = {english}
```
Comment (JimMadge) [inline book/website/_bibliography/references.bib:2683]:
```suggestion
```

File: book/website/_bibliography/references.bib [hunk 2]
```diff
@@ -2662,3 +2662,41 @@ @misc{WEF2022commons
...
+  date = {2012},
+  year = {2012},
+  publisher = {arXiv},
+  journal = {arXiv},
+  doi = {10.48550/ARXIV.1201.0490},
+  url = {https://arxiv.org/abs/1201.0490},
+  urldate = {2024-06-21},
+  abstract = {Scikit-learn is a Python module integrating a wide range of state-of-the-art machine learning algorithms for medium-scale supervised and unsupervised problems. This package focuses on bringing machine learning to non-specialists using a general-purpose high-level language. Emphasis is put on ease of use, performance, documentation, and API consistency. It has minimal dependencies and is distributed under the simplified BSD license, encouraging its use in both academic and commercial settings. Source code, binaries, and documentation can be downloaded from http://scikit-learn.org.},
```
Comment (JimMadge) [inline book/website/_bibliography/references.bib:2697]:
```suggestion
```

File: book/website/_bibliography/references.bib [hunk 3]
```diff
@@ -2662,3 +2662,41 @@ @misc{WEF2022commons
...
+  version = {4},
+  keywords = {FOS: Computer and information sciences,Machine Learning (cs.LG),Mathematical Software (cs.MS)}
```
Comment (JimMadge) [inline book/website/_bibliography/references.bib:2699]:
```suggestion
```

**Review by JimMadge — CHANGES_REQUESTED** [review]
Looks good.

Assuming we can do without the abstracts and some other fields in the bibliography.

**Comment by RichardJActon** [comment]
Yes - thanks @JimMadge (just copy pasted from my reference manager's output and didn't think to remove the bloat)

**Review by malvikasharan — APPROVED** [review]

**Review by JimMadge — APPROVED** [review]

**Pull request merged**